Realm.Shared/Realm.cs

```diff
@@ -431,18 +431,36 @@ private RealmObject CreateObject(string className, out RealmObject.Metadata meta
             return result;
         }
 
-        internal RealmObject MakeObjectForRow(string className, RowHandle rowHandle)
+        internal RealmObject MakeObjectForRow(RealmObject.Metadata metadata, IntPtr rowPtr)
```

Comment:
Fair point - the overload taking a metadata and rowHandle was just added in this merge as part of fixing a bug in RealmResults.